src/trace-forwarder/src/logging.rs

```diff
@@ -0,0 +1,432 @@
+// Copyright (c) 2019-2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use slog::{o, record_static, BorrowedKV, Drain, Key, OwnedKV, OwnedKVList, Record, KV};
+use std::collections::HashMap;
+use std::io;
+use std::io::Write;
+use std::process;
+use std::result;
+use std::sync::Mutex;
+
+const LOG_LEVELS: &[(&str, slog::Level)] = &[
+    ("trace", slog::Level::Trace),
+    ("debug", slog::Level::Debug),
+    ("info", slog::Level::Info),
+    ("warn", slog::Level::Warning),
+    ("error", slog::Level::Error),
+    ("critical", slog::Level::Critical),
+];
+
+// XXX: 'writer' param used to make testing possible.
+pub fn create_logger<W>(name: &str, source: &str, level: slog::Level, writer: W) -> slog::Logger
+where
+    W: Write + Send + Sync + 'static,
```

Comment:
Why does `W` need to be `Send + Sync + 'static`? Do you have multiple threads and you need to send this writer over?

Reply:
Hi @sboeuf. First, I've removed that file since it was essentially a copy of the existing `logging.rs` in the new agent. I've now moved that code to a `pkg/` directory so the two components can share it.

As to why those traits are required - it's a requirement of the `slog-async` crate we're already using in the agent.

Reply:
Got it!